docker/Dockerfile.centos.7

```diff
@@ -9,43 +43,76 @@
 FROM centos:7
 MAINTAINER Johann Lombardi <johann.lombardi@intel.com>
 
-# Build arguments can be set via -build-arg
-# set NOBUILD to disable git clone & build
-ARG NOBUILD
+# Build arguments can be set via --build-arg
 # use same UID as host and default value of 1000 if not specified
 ARG UID=1000
+ARG USE_RPMS=false
 
-ENV USER daos
-ENV PASSWD daos
+# for good measure, clean the metadata
+RUN yum clean metadata
 
 # Install basic tools
-RUN yum install -y epel-release yum-plugin-copr; \
-  yum -y copr enable jhli/ipmctl; \
-  yum -y copr enable jhli/safeclib; \
-  yum install -y  boost-devel clang-analyzer cmake CUnit-devel doxygen flex \
-    gcc gcc-c++ git golang graphviz make lcov meson nasm numactl-devel \
-    ninja-build openssl-devel readline-devel pandoc patch scons sg3_utils \
-    valgrind valgrind-devel yasm \
-    libaio-devel libcmocka libcmocka-devel libevent-devel libibverbs-devel \
-    libiscsi-devel librdmacm-devel libtool libtool-ltdl-devel \
-    libuuid-devel libyaml-devel \
-    python python-jwt python-pep8 python-virtualenv \
-    python2-pip python2-pygithub python34-pip python34-pygithub pylint \
-    libipmctl-devel
-
-RUN useradd -u $UID -ms /bin/bash $USER ;echo "$USER:$PASSWD" | chpasswd
+RUN yum -y install epel-release
+RUN yum -y install                                                 \
+           git gcc gcc-c++ make cmake golang libtool boost-devel   \
```

Comment:
A sort of alphabetizing the package list helps you see at a glance if a package is missing or duplicated in the list of packages.

I generally put the lib* and python* packages in their own set of lines since there are quite a few of them.

Reply:
Own set of lines is OK as long as it's not their own `RUN` lines, to keep the docker image count reasonable.

But yes, some sorting in the future would be useful.  Right now I am just trying to get all of the Dockerfiles for a given OS as harmonized as I can on the most set of packages so that we are getting the most effective and efficient use out of the cache.